shared/src/telemetry/data.rs

```diff
@@ -0,0 +1,55 @@
+use serde_json::json; // Ensure you have `serde_json` in your Cargo.toml
+use serde_json::Value;
+use std::env;
+use std::fs;
+use std::path::PathBuf;
+
+// Function to get the PostgreSQL data directory from the PGDATA environment variable
+pub fn get_postgres_data_directory() -> Option<String> {
+    env::var("PGDATA").ok()
+}
+
+pub fn read_telemetry_data(extension_name: String) -> Result<Value, String> {
+    // Determine the base PostgreSQL data directory
+    let pg_data_directory =
+        get_postgres_data_directory().ok_or("PGDATA environment variable is not set")?;
+
+    // Customize the path based on the extension name
+    let directory_path = PathBuf::from(pg_data_directory).join(match extension_name.as_str() {
+        "pg_bm25" => "paradedb",
+        "pg_analytics" => "deltalake",
+        _ => return Err("Unknown extension name".to_string()),
+    });
+
+    // Convert the PathBuf back to a String for use in error messages and JSON object
+    let directory_path_str = directory_path
+        .to_str()
+        .ok_or("Failed to convert directory path to string")?;
+
+    // Get the metadata for the specified directory
+    let dir_metadata = fs::metadata(&directory_path)
+        .map_err(|e| format!("Error getting metadata for {}: {}", directory_path_str, e))?;
+
+    // Check if the specified path is a directory
+    if !dir_metadata.is_dir() {
+        return Err(format!("{} is not a directory", directory_path_str));
+    }
+
+    // Calculate total size of files in the directory
+    let dir_size: u64 = fs::read_dir(&directory_path)
```

Comment:
This is a bit hard to read. Also, does this handle the case where you have folders nested within folders?